internal/daemon/daemon.go

```diff
@@ -47,7 +47,7 @@ func NewDaemon(mode string, client client.Client, vspImages map[string]string, c
 	}
 }
 
-func (d *Daemon) Run() error {
+func (d *Daemon) ListenAndServe() error {
```

Comment:
Is this a little confusing? Now the daemon will have 2 methods ListenAndServer differentiated only by the return values IIUC?

Maybe daemon.ListenAndServer() should be  renamed StartListenAndServer or something more specific.

Or Run->InitDaemon or something.

Reply:
The refactor is not complete yet. The goal is to have two functions:

Listen
and
Serve.

Listen always returns a socket that needs to be passed to Serve. Any pre-init required before Serve blocks is done in Listen so that the design becomes composable. That means that you will be able to use mulitple SideManagers in Daemon. Daemon itself will also have a Listen and a Serve. The intent remains the same, call Listen recursively inside the top level Listen of daemon to init all SideManagers returning one of the sockets and pass to back in to Serve.

Following this design, it will be possible to test the hardware detection (and the DPU CRs) in a race-condition free way.